ios/Classes/SwiftTwilioVoicePlugin.swift

```diff
@@ -865,12 +865,14 @@ public class SwiftTwilioVoicePlugin: NSObject, FlutterPlugin,  FlutterStreamHand
         }
     }
     
-    func reportIncomingCall(from: String, uuid: UUID) {
-        let callHandle = CXHandle(type: .generic, value: from)
+    func reportIncomingCall(from: String, uuid: UUID) {        
+        // Using .phoneNumber to fetch display name from the Phonebook if the number is saved otherwise show literal string 'from'.
+        let callHandle = CXHandle(type: .phoneNumber, value: from)  
         
         let callUpdate = CXCallUpdate()
         callUpdate.remoteHandle = callHandle
-        callUpdate.localizedCallerName = clients[from] ?? self.clients["defaultCaller"] ?? defaultCaller
+        /// Apparently localizedCallerName overrides remoteHandle, so it was commented out fix "Unknown Caller" issue.
+        //  callUpdate.localizedCallerName = from ?? self.clients["defaultCaller"] ?? defaultCaller
```

Comment:
This won't respect saved clients, integrating this change will affect others making use of the `registerClient` feature.

I'm working on an different approach allowing the developer to interpret the name directly from the custom parameters, clients map, etc.